Борзов Артем

qoter opened the pull request.

qoter committed:
fuuuuuf

qoter committed:
pingggg

honest-hrundel commented:
:tomato: Не пройден линтинг или базовые тесты

qoter committed:
hrundel

'

honest-hrundel commented:
:tomato: Не пройден линтинг или базовые тесты

qoter committed:
eslint fix spread

honest-hrundel commented:
:green_apple: Пройден линтинг и базовые тесты

Superilya commented on .eslintrc:
```diff
@@ -3,5 +3,10 @@
+    },
+    "parserOptions":{
+        "ecmaFeatures": {
+            "experimentalObjectRestSpread": true
```
👍

Superilya commented on middlewares/weather.js:
```diff
@@ -0,0 +1,21 @@
+
+module.exports = async (req, res, next) => {
+    let location;
+    if (req.query.query) {
```
зачем эта логика в middleware? вынеси её нормально в контроллер он для этого и нужен. Если тебе надо переиспользовать код делай базовую модель, от неё наследуйся и переиспользуй сколько угодно

Superilya replied:
или базовый контроллер. Использовать модели в мидлваре нельзя

qoter replied:
Выпилил)

Superilya commented on models/news.js:
```diff
@@ -0,0 +1,24 @@
+        return newsapi.v2.topHeadlines({
+            country: country,
+            category: category
+        }).then(res => res.articles)
```
это 🎩, надо юзать async/await

Superilya commented on models/weather.js:
```diff
@@ -0,0 +1,56 @@
+
+    static loadConsolidated(location) {
+        return getJson(`location/${location.woeid}`)
+            .then(res => res.consolidated_weather)
```
зачем делать then, есть есть async/await 🤷‍♀️

Superilya commented on public/categories.css:
```diff
@@ -0,0 +1,38 @@
+.categories {
```
хороший тон, прогнать svg через svgo, лучше делать это при сборке проекта. Кароч тебе надо знать что svg надо обрабатывать перед отправкой клиенту, если нет автоматизации для . этого придётся сделать это руками

qoter replied:
Прогнал)

Superilya commented on models/weather.js:
```diff
@@ -0,0 +1,56 @@
+];
+
+class Weather {
+    constructor(city, temp, windSpeed, date, weatherStateAbbr) { // eslint-disable-line max-params
```
max-params не просто так, может подумать над тем, что бы переделать на один объект типа options?

Superilya commented on models/weather.js:
```diff
@@ -0,0 +1,56 @@
+    'Декабря'
+];
+
+class Weather {
```
очень странный класс, это какой то паттерн? почему в loadConsolidated не вернуть просто объекты а надо возвращать инстансы самого себя? очень не явная логика

qoter replied:
Ну логично, что статический метод загружающий погоду, возвращает инстансы погоды.
Если, например, апи поменяется, то не придётся менять все использования.

qoter committed:
1. Svgo
2. Aync/await вместо then
3. Убрал weatherMiddleware
4. Выпилил grid, вёрстка стала проще
5. Конфиг

honest-hrundel commented:
:tomato: Не пройден линтинг или базовые тесты

qoter committed:
fix hbs could not found problem

honest-hrundel commented:
:tomato: Не пройден линтинг или базовые тесты

Commit:
hardcode apikey

honest-hrundel commented:
:tomato: Не пройден линтинг или базовые тесты

Commit:
remove config module

honest-hrundel commented:
:green_apple: Пройден линтинг и базовые тесты

Superilya commented on models/weather.js:
```diff
@@ -0,0 +1,75 @@
+const config = require('../config/default');
+const fetch = require('node-fetch');
+
+const API_URI = 'https://www.metaweather.com/api/';
```
уноси всё в конфиг в следующий раз
